packages/now-build-utils/src/detectors/hugo.ts

```diff
@@ -0,0 +1,18 @@
+import { DetectorParameters, DetectorResult } from '../types';
+
+export default async function detectHugo({
+  fs: { exists },
+}: DetectorParameters): Promise<DetectorResult> {
+  const [hasConfigYaml, hasConfigToml] = await Promise.all([
+    exists('config.yaml'),
+    exists('config.toml'),
+  ]);
+  if (!hasConfigYaml && !hasConfigToml) {
+    return false;
+  }
+  return {
+    buildCommand: ['hugo'],
+    buildDirectory: 'public',
```

Comment:
You need to read the config file to detect the output directory called `publishDir`.

https://gohugo.io/getting-started/configuration/